crates/ty_python_semantic/resources/mdtest/narrow/type.md

```diff
@@ -127,7 +127,8 @@ class B: ...
 
 def _[T](x: A | B):
     if type(x) is A[str]:
-        reveal_type(x)  # revealed: (A[int] & A[Unknown]) | (B & A[Unknown])
+        # TODO: previous was revealed: (A[int] & A[Unknown]) | (B & A[Unknown])
+        reveal_type(x)  # revealed: Never
```

Comment:
@carljm this is the one change in the tests I'm quite unsure of.

I'm not sure why the previous revealed type wasn't `(A[int] & A[str]) | (A[str] & B)`. I also think that first branch should reduce to `Never` even if `A` is bivariant, but the second still seems possible, so not sure why the clause is being detected as unreachable.